### PR TITLE
add 'incomplete-read/read1' tests

### PR DESCRIPTION
**Pull request by abebeos**
- re #3250

PR https://github.com/urllib3/urllib3/pull/3186 introduced support for the read1() method in class HTTPResponse.

The added tests showcase that the current implementation of `read1()` is inconsistent.

Depending on data, `DecodeError` is either raised or not. It should be raised always (and all tests should pass, as with `read()`)

```sh
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[f] PASSED                                 [ 25%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[fo] FAILED                                [ 25%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[foo] FAILED                               [ 26%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[foofoofoofoofoo] PASSED                   [ 26%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[123456789012345] FAILED                   [ 26%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[foo4] FAILED                              [ 26%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx] PASSED
```

(please focus within this PR only on the tests themselves. If they are correct, then they should be merged, possibly with an 'ignore/skip' marker for the failing test)

### CODE REVIEW

File: test/test_response.py [hunk 1]
```diff
@@ -448,6 +448,45 @@ def test_decode_zstd_incomplete(self, data: bytes) -> None:
...
+        data = zstd.compress(data)
+        fp = BytesIO(data[:-1])
+
+        r = HTTPResponse(
+            fp, headers={"content-encoding": "zstd"}, preload_content=False
+        )
+
+        # read/decode via read1(!), expecting DecodeError
+        with pytest.raises(DecodeError):
+            r.read1(decode_content=True)
```
Comment (illia-v) [inline test/test_response.py:488]: Thanks for checking the new `read1` functionality!

In this case, I am not convinced there is a bug, the tests look incorrect. You can get this test to stop failing if you add one more `r.read1(decode_content=True)` call. Similarly, you can make `test_decode_zstd_incomplete_read` fail if you call `r.read(amt=len(data), decode_content=True)`.

`read1` is not guaranteed to read all the data. If it reads a piece of the compressed data and successfully decodes the data, it returns it without flushing the decoder. This is just like `read` when passed `amt` is less than the total content length.

In this case, `r.read1(decode_content=True)` returns `data[:-1]`.

Comment (abebeos) [reply test/test_response.py:488]: I think i've understood.

**Comment by abebeos** [comment]
@illia-v, could you please reopen so i can try to adjust the tests?

**Comment by abebeos** [comment]
follow-up: https://github.com/urllib3/urllib3/pull/3265

**Pull request closed without merging**